plugins/inputs/redis_consumer/redis_consumer.go

```diff
@@ -0,0 +1,170 @@
+package redis_consumer
+
+import (
+	"fmt"
+	"net/url"
+	"strings"
+	"sync"
+
+	"github.com/garyburd/redigo/redis"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"github.com/influxdata/telegraf/plugins/parsers"
+)
+
+type RedisConsumer struct {
+	Servers  []string
+	Channels []string
+
+	parser parsers.Parser
+
+	pubsubs []redis.PubSubConn
+	sync.Mutex
+	acc telegraf.Accumulator
+}
+
+var sampleConfig = `
+  ## Specify servers via a url matching:
+  ##  [protocol://][:password]@address[:port]
+  ##  e.g.
+  ##    tcp://localhost:6379
+  ##    tcp://:password@192.168.99.100
+  ##
+  ## If no servers are specified, then localhost is used as the host.
+  ## If no port is specified, 6379 is used
+  servers = ["tcp://localhost:6379"]
+
+  ##  List of channels to listen to. Selecting channels using pattern-matching
```

Comment:
what type of pattern-matching? Is it regex or something redis-specific? Can you give a short example?